src/utils/tree.js

```diff
@@ -0,0 +1,108 @@
+import { Foldable, foldl, foldr, Functor, map } from 'funcadelic';
+import { reduceObject } from 'ioo';
+
+import defineComputedProperty from './define-computed-property';
+import getReducerType from './get-reducer-type';
+import isList from './is-list';
+import isPrimitive from './is-primitive';
+import propertiesFor from './properties-for';
+import transitionsFor from './transitions-for';
+
+class TypeTree {
+  constructor() {
+    this.data = null;
+  }
+  static from(Type, path = []) {
+    let primitive = isPrimitive(Type);
+    let [name] = path.slice(-1);
+    let list = isList(Type);
+    return Object.create(TypeTree.prototype, {
+      data: {
+        get() {
+          return {
+            name,
+            path,
+            type: getReducerType(Type),
+            schemaType: Type,
+            isPrimitive: primitive,
+            isComposed: !primitive,
+            isList: list,
```

Comment:
Where is this information used?

Reply:
It might not be anymore. It was used with parameterized array